core/src/main/java/ai/timefold/solver/core/api/domain/variable/CascadingUpdateShadowVariable.java

```diff
@@ -3,54 +3,24 @@
 import static java.lang.annotation.ElementType.FIELD;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
-import java.lang.annotation.Repeatable;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
-import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
-
 /**
- * Specifies that field may be updated by the target method when one or more source variables change.
+ * Specifies that field may be updated by the target method when a dependency changes.
  * <p>
- * Automatically cascades change events to {@link NextElementShadowVariable} of a {@link PlanningListVariable}.
+ * Automatically cascades change events to the next elements of a {@link PlanningListVariable}.
  * <p>
  * Important: it must only change the shadow variable(s) for which it's configured.
- * It is only possible to define either {@code sourceVariableName} or {@code sourceVariableNames}.
  * It can be applied to multiple fields to modify different shadow variables.
  * It should never change a genuine variable or a problem fact.
  * It can change its shadow variable(s) on multiple entity instances
  * (for example: an arrivalTime change affects all trailing entities too).
  */
 @Target({ FIELD })
 @Retention(RUNTIME)
-@Repeatable(CascadingUpdateShadowVariable.List.class)
 public @interface CascadingUpdateShadowVariable {
```

Comment:
The Javadoc should also discuss the differences between this and a shadow variable.
Specifically, that nothing is allowed to depend on it, and that it will always be processed after all other shadow variables have been updated.

Maybe it should discuss what happens if multiple cascading vars on the same entity instance depend on each other. (Because they share fields.)